api/v1alpha1/dataset_types.go

```diff
@@ -347,7 +347,7 @@ func (dataset *Dataset) GetDataOperationInProgress(operationType string) string
 
 // SetDataOperationInProgress set the data operation running on this dataset,
 func (dataset *Dataset) SetDataOperationInProgress(operationType string, name string) {
-	if dataset.Status.OperationRef == nil {
+	if dataset.Status.OperationRef == nil || dataset.Status.OperationRef[operationType] == "" {
 		dataset.Status.OperationRef = map[string]string{
```

Comment:
The `OperationRef[operationType]` nil check is wrong under the following case:

1. dataload A initialize the OperationRef from nil to {"dataload": "A"};
2. datamigrate B the will set the OperationRef to {"datamigrate": ",B"}, the dataload 'A' is missed;

Reply:
fixed in https://github.com/fluid-cloudnative/fluid/pull/3261